Const correctness on RunTimeOptions

nhuang-tt opened the pull request:
### Ticket
N/A

### Problem description
- Getters should be const to prevent accidental mutation
- Let people with a const reference to access these getters

### What's changed
Make them const

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

omilyutin-tt commented on tt_metal/api/tt-metalium/rtoptions.hpp:
```diff
@@ -167,52 +167,53 @@ class RunTimeOptions {
     const std::string& get_kernel_dir() const;
 
-    inline bool get_build_map_enabled() { return build_map_enabled; }
+    inline bool get_build_map_enabled() const { return build_map_enabled; }
```
Can drop `inline` everywhere too :) 

It is implied in this context

nhuang-tt committed:
Const correctness on RunTimeOptions

Getters should be const to prevent accidental
mutation to the options, and also to allow users with
a const reference to access them

omilyutin-tt commented on tt_metal/llrt/rtoptions.hpp:
```diff
@@ -147,118 +147,113 @@ class RunTimeOptions {
     RunTimeOptions(const RunTimeOptions&) = delete;
     RunTimeOptions& operator=(const RunTimeOptions&) = delete;
 
-    inline bool is_root_dir_specified() const { return this->is_root_dir_env_var_set; }
```
83 `inline` removed! This is amazing🎖️
